Neos.Flow/Classes/Security/Authentication/AuthenticationProviderManager.php

```diff
@@ -293,99 +292,12 @@ protected function emitLoggedOut()
     }
 
     /**
-     * Builds the provider and token objects based on the given configuration
+     * Signals that authentication commenced and at least one token was authenticated.
      *
      * @return void
-     * @throws Exception\InvalidAuthenticationProviderException
-     * @throws Exception\NoEntryPointFoundException
+     * @Flow\Signal
      */
-    protected function buildProvidersAndTokensFromConfiguration()
+    protected function emitSuccessfullyAuthenticated()
```

Comment:
Is that new? If so I'd vote for moving that to a separate PR in order to keep this one a pure refactoring and to make it easer to review, test & debug

Reply:
This is needed together with the connections in the `Context` (see your comment from Package.php) in order to separate AuthenticationManager and Context and not have them cross depend on each other. It's the biggest part of this refactoring while everything else is really just moving code around.

Ofc I could try to split that into two PRs, if you really think that's needed.

Reply:
> I could try to split that into two PRs, if you really think that's needed

I really don't want to decide that for you. It's just a concern I had when doing a first review: mixing behavioral changes into refactorings has introduced a lot of regressions in the past. But if it's a requirement for a clean separation I'm the last to block this :)